app/code/community/Inviqa/SymfonyContainer/Model/Observer.php

```diff
@@ -1,12 +1,40 @@
 <?php
-
+use Inviqa_SymfonyContainer_Model_ControllerInjectionCompilerPass as ControllerInjectionCompilerPass;
 
 class Inviqa_SymfonyContainer_Model_Observer
 {
+    use Inviqa_SymfonyContainer_Helper_ServiceProvider;
+
     public function onCacheRefresh(Varien_Event_Observer $event)
     {
         if (Inviqa_SymfonyContainer_Model_ConfigurationBuilder::MODEL_ALIAS === $event->getType()) {
             unlink(Mage::getBaseDir('cache') . '/' . Inviqa_SymfonyContainer_Model_ConfigurationBuilder::CACHED_CONTAINER);
         }
     }
-}
\ No newline at end of file
+    public function onPreDispatch(Varien_Event_Observer $event)
+    {
+        $controller = $event->getControllerAction();
+        $controllerServices = $this->getService(ControllerInjectionCompilerPass::CONTROLLERS_SERVICE_ID)->controllers;
+
+        if (!isset($controllerServices[get_class($controller)])) {
+            return;
+        }
+
+        if (!in_array('setDependencies', get_class_methods($controller))) {
```

Comment:
because its automatically called by the __construct method :( - so it will get called twice
(nothing wrong in principle with it being called twice, once with no args and once with services, but it might be a bit confusing)